File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
  def removeNthFromEnd(head: ListNode, n: Int): ListNode = {
    var f = head
    for (_ <- 0 to n) {
      if (f == null) return head.next
      else f = f.next
    }
    var s = head
    while (f != null) {
      f = f.next
      s = s.next
    }
    s.next = s.next.next
    head
  }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
  tail := head
  for i := 0; i < n; i++ {
    tail = tail.Next
  }
  var pre *ListNode
  ans := head
  for ; tail != nil; tail = tail.Next {
    pre = ans
    ans = ans.Next
  }
  if ans == head {
    return head.Next
  } else {
    pre.Next = ans.Next
    return head
  }
}